libhfcommon/util.c

```diff
@@ -140,6 +143,33 @@ bool util_PinThreadToCPUs(uint32_t threadno, uint32_t cpucnt) {
         return false;
     }
     return true;
+#elif defined(__sun)
+    psetid_t set;
+    pid_t p;
+    bool ret = true;
+
+    if (pset_create(&set) != 0) {
+        PLOG_W("pset_create failed");
+        return false;
+    }
+
+    for (uint32_t i = 0; i < cpucnt; i++) {
+        if (pset_assign(set, i, NULL) != 0) {
```

Comment:
I think it should be 

```(start_cpu + i) % num_cpus``` instead of just ```i```

Otherwise all threads will be pinned to the same CPU? Also ```% num_cpus``` to make sure that CPU cores actually exist. 

BTW, Thanks for the patches.